### ISSUE TITLE
Primary API documentation is incomplete or insufficient 

### ISSUE THREAD

**Issue by AdaDoom3** [omcljs/om#365]
The API documentation wiki for Om lacks accessibility to programmers new to Clojure.
https://github.com/omcljs/om/wiki/Documentation

For example, **path**, **state**, and **value** have no explanation, and function parameters' purposes are not often described - like in **build-all** the **x** parameter is summed up only with *"x can be any value"*.

**Comment by JulianLeviston** [comment]
No offense intended, but I don't actually think Om (or any web framework, for that matter) is a great fit for programmers new to Clojurescript. Pehaps you could create a bridging tutorial called "Om for programmers new to Clojurescript" that teaches ClojureScript to new people in the context of Om. The trouble is there's so much to learn even in the basic Om tutorials. Documentation needs to be written for those who understand (it's reference material, after all).

Or, they could just use the getting started introductory tutorial (https://github.com/omcljs/om/wiki/Basic-Tutorial), which itself recommends if you're not familiar with Clojure or Clojurescript that you do the Clojurescript tutorial for Lighttable first. (https://github.com/swannodette/lt-cljs-tutorial).

Hope that helps! :)

**Comment by AdaDoom3** [comment]
No problem Julian this is definitely a worthwhile topic to debate.

I think there is really no decent reason that the 30 or so main functions don't have a little more information (I mean it is *the* official documentation) so that even experienced programmers don't have to thumb through 4 tutorials to find some example calls. Case in point: can you find one example on the internet of **detach-root**?

Clearly, it may be a waste of time to be really basic, but I am sure experienced programmers could benefit from additional information. I mean, the example I gave in the earlier comment essentially described the type of **x** without describing its purpose... It would take a person who thoroughly understands the library a day at most to do and end some user frustration.

**Comment by bensu** [comment]
I have a [branch](https://github.com/bensu/om/tree/docs) with auto generated docs by [codeina](https://github.com/funcool/codeina). If @swannodette is on board I wouldn't mind doing what's left:

* Taking the [Documentation](https://github.com/omcljs/om/wiki/Documentation) and adding it to the docstrings.
* Adding docstrings in `om.dom` to the autogenerated functions/macros.
* Hiding the `om.next` namespaces from the docs.

Note: my branch is not ready since it is waiting on this [PR](https://github.com/funcool/codeina/pull/3)

**Comment by swannodette** [comment]
Not interested in in anything based on Codox. ClojureScript has good support for JSDoc in docstrings now and thus not far from YUIDoc support. Somebody would need to make a YUIDoc template for ClojureScript projects, but that's the only thing I'm interested in at this point.
